### PR TITLE
Migrate Team.GetActiveMembersCount to Sync by default

### LINKED ISSUE #11119: Migrate "Team.GetActiveMemberCount" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-16180)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `GetActiveMemberCount` in the `Team` store.

The expected way to implement it is, go to the `Team` store implementation in the `store/sqlstore/` directory, modify the method `GetActiveMemberCount` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by rvillablanca**
Resolves #11119

### CODE REVIEW

File: app/team.go [hunk 1]
```diff
@@ -1165,7 +1165,7 @@ func (a *App) RestoreTeam(teamId string) *model.AppError {
 
 func (a *App) GetTeamStats(teamId string) (*model.TeamStats, *model.AppError) {
 	tchan := a.Srv.Store.Team().GetTotalMemberCount(teamId)
-	achan := a.Srv.Store.Team().GetActiveMemberCount(teamId)
+	aCount, aErr := a.Srv.Store.Team().GetActiveMemberCount(teamId)
```
Comment (mounicapaladugu) [inline app/team.go]: you can probably have the variable names as count, err here?

File: store/sqlstore/team_store.go [hunk 2]
```diff
@@ -682,13 +681,11 @@ func (s SqlTeamStore) GetActiveMemberCount(teamId string) store.StoreChannel {
 				AND TeamMembers.TeamId = :TeamId
 				AND TeamMembers.DeleteAt = 0
 				AND Users.DeleteAt = 0`, map[string]interface{}{"TeamId": teamId})
-		if err != nil {
-			result.Err = model.NewAppError("SqlTeamStore.GetActiveMemberCount", "store.sql_team.get_member_count.app_error", nil, "teamId="+teamId+" "+err.Error(), http.StatusInternalServerError)
-			return
-		}
+	if err != nil {
+		return 0, model.NewAppError("SqlTeamStore.GetActiveMemberCount", "store.sql_team.get_member_count.app_error", nil, "teamId="+teamId+" "+err.Error(), http.StatusInternalServerError)
```
Comment (mounicapaladugu) [inline store/sqlstore/team_store.go]: can you change error message store.sql_team.get_member_count.app_error --> store.sql_team.get_active_member_count.app_error, just to be consistent

File: app/team.go (hunk 1 above)
Comment (rvillablanca) [reply app/team.go]: Ok

File: store/sqlstore/team_store.go (hunk 2 above)
Comment (rvillablanca) [reply store/sqlstore/team_store.go]: OK

Comment (mounicapaladugu) [reply store/sqlstore/team_store.go]: just these 2 minor changes from my end. rest all look great!

Comment (rvillablanca) [reply store/sqlstore/team_store.go]: Both changes are done @mounicapaladugu, but It seems you have problems with Jenkins, sometimes if fails with minor changes

File: app/team.go [hunk 3]
```diff
@@ -1165,7 +1165,7 @@ func (a *App) RestoreTeam(teamId string) *model.AppError {
 
 func (a *App) GetTeamStats(teamId string) (*model.TeamStats, *model.AppError) {
 	tchan := a.Srv.Store.Team().GetTotalMemberCount(teamId)
-	achan := a.Srv.Store.Team().GetActiveMemberCount(teamId)
```
Comment (jespino) [inline app/team.go]: We were doing it concurrently (actually it keeps been concurrent execution) but I think i prefer to use the explicit async call here (with the channel and the gorouting, see the example PR).

File: store/sqlstore/team_store.go [hunk 4]
```diff
@@ -669,9 +669,8 @@ func (s SqlTeamStore) GetTotalMemberCount(teamId string) store.StoreChannel {
 	})
 }
 
-func (s SqlTeamStore) GetActiveMemberCount(teamId string) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		count, err := s.GetReplica().SelectInt(`
+func (s SqlTeamStore) GetActiveMemberCount(teamId string) (int64, *model.AppError) {
+	count, err := s.GetReplica().SelectInt(`
 			SELECT
```
Comment (jespino) [inline store/sqlstore/team_store.go]: nit: Probably 1 less tab for each line of the query string would make it looks better, and probably, moving the entire query to a query variable can be good too.

Comment (rvillablanca) [reply store/sqlstore/team_store.go]: Are you sure about a new query var is needed here ? The others methods follow this pattern. 
As to the tab for each line I think is ok

**Comment by rvillablanca** [comment]
@jespino I have finished all your requested changes

**Comment by jespino** [comment]
@rvillablanca because you changed (correctly) an error string, you need to run `make i18n-extract` and add the correct text of the new error string to the `i18n/en.json` file. you can base the text in the existing text for the `store.sql_team.get_member_count.app_error` key.

**Comment by rvillablanca** [comment]
I run the target but it no only add the new key, also it removed others ones. I suppose it is ok

**Comment by jespino** [comment]
@rvillablanca Sorry, yes and no, you don't have the enterprise repo so... you can't use "make i18n-extract" properly. Just pick only the added line, add the text, and do not remove the rest of the lines.

**Comment by rvillablanca** [comment]
@jespino ok, and it only applies to en.json ? what about spanish or another lang ?

**Comment by jespino** [comment]
@rvillablanca only to the en.json, the rest of the languages are translated in https://translate.mattermost.com/ (You can help there if you want too), and then we overwrite the translations files for the rest of the languages in the repo, so... do not changes other i18n/*.json files in PRs.

**Comment by jespino** [comment]
@rvillablanca Now LGTM, Thanks!

**Pull request merged**